src/components/Feedback/FileProgress/FileProgress.tsx

```diff
@@ -0,0 +1,72 @@
+import { FC, useMemo } from 'react';
+import { FileWithPath } from 'react-dropzone';
+
+import CompactFileProgress from './CompactFileProgress';
+import RegularFileProgress from './RegularFileProgress';
+
+interface FileProgressBaseProps {
+  onDelete: () => void;
+  file: FileWithPath | File;
+  isDone?: boolean;
+  progressPercent?: number;
+  onCancel?: () => void;
+  isError?: boolean;
+}
+
+export interface RegularFileProgressBaseProps extends FileProgressBaseProps {
+  compact?: false;
+  fullErrorText?: string;
+  customLoadingText?: string;
+  customCompleteText?: string;
+  onRetry?: () => void;
+}
+export interface CompactFileProgressBaseProps extends FileProgressBaseProps {
+  compact: true;
+  shortErrorText?: string;
+  fullErrorText?: string;
+}
+
+export interface FileProgressPropsExtension {
+  showCompleteState: boolean;
+  handleOnClick: () => void;
+}
+
+const FileProgress: FC<
+  RegularFileProgressBaseProps | CompactFileProgressBaseProps
+> = (props) => {
+  const showCompleteState = useMemo(() => {
+    if (props.isError) return true;
+    if (props.isDone === undefined) {
```

Comment:
Could these be merged into 1 if-statement? `if (props.isError || props.isDone === undefined)`